backend/pythonAPI/openPipePy.py

```diff
@@ -1,10 +1,163 @@
+
+  def addMetaTag(self, metaDataId, tagName, value):
+    if metaDataId is None or tagName is None or value is None:
+        return({"result":"Fail"}))
+
+    else:
+        return (BL().insertMetaTag(metaDataId,tagName,value))
+
+ def createMetaData(self)
+          return{"result": BL().insertIntoMetaData()}
+
+def deleteCanonicalMetaTag(self, id):
+      if id is None:
+        return {"result":"Fail"}
+      else:          
+        return (deleteTag(dict["id"])
+
+  def deleteMetaTag(self,metaDataId,tagName,value):
+    if metaDataId is None or tagName is None or value is None:
+       return {"result":"Fail"}
+    else:          
+        return (BL().deleteMetaTag(metaDataId,tagName,value))
+
+ def getAssetMetaTags(self,assetId):
+        if assetID is None:
+          return BL().getAssetMetaTags(dict['assetId'])
+
+  def getCanonicalMetaTags(self):
+    return (getAllTags())
+
+  def updateMetaTag(self, metaDataId, oldTagName,oldValue,newTagName,newValue):
+      if metaDataId is None or oldTagName is None or oldValue is None or newTagName is None or newValue is None:
+           return {"result":"Fail"}
+      else:  
+          return BL().updateMetaTag(dict["metaDataId"],dict["oldTagName"],dict["oldValue"],dict["newTagName"],dict["newValue"])}))
+
+
+   def updateCanonicalMetaTag(self,id):
+    if id is None or name is None:
+      return ({"result":"Fail"})
+    else:
+      return (updateTag(dict["id"],dict["name"]))
+        
+   # **********************************************************************************************
+    # ************************************** Collection ************************************************
+    # **********************************************************************************************
+
+    def  createCollection(self,name):
+      if name is None:
+        return{"result": "Fail"}
+      else:
+       return {'result':BL().insertIntoCollection(dict["name"])}
+
+
+
+ def getCollections(self, start, end):
+      if start is  and end is:
+        return (BL().getRangeOfCollections(dict["start"],dict["end"])
+      else: 
+        if collectionId is None:
+          return ({"total": [""], "data": [{"id": [""], "name": [""], "timeStamp": [""]}]})
+        else:   
+           if id == -1:
+             return {"total": [""], "data": [{"id": [""], "name": [""], "timeStamp": [""]}]}
+           elif id == "all" :
+              if limit in :
+                return(BL().getAllCollections(dict["limit"]))
+              else:
+                return(BL().getAllCollections(-1))
+            else:
+             return(BL().getCollectionByID(id))
+
+
+
+
+  def getPublicAssetsInCollection(self, p,ps,collectionId):
+    if p is None:
+      return "1";
+    if ps is None:
+      return "10";
+    if collectionId is None:
+      return ({"total": "-1", "data": [{}]})
+    else:
+     return (BL().getPublicAsssetsInCollection(dict['collectionId'],int(dict["p"]), int(dict["ps"])))
+          
+
+ 
```

Comment:
The collection is the same as folder in some endpoints it has not been refactored yet.please delete this part